specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json

```diff
@@ -1620,6 +2077,166 @@
+                  "type": "string"
+                },
+                "pythonVersion": {
+                  "description": "The Python version used for Spark. This property is mutually exclusive with containerSettings.",
+                  "type": "string",
+                  "enum": [
+                    "2",
+                    "3"
+                  ],
+                  "default": "3"
+                },
+                "containerSettings": {
+                  "title": "Container settings.",
+                  "description": "Custom docker container settings for the Spark cluster. This property is mutually exclusive with sparkVersion and pythonVersion.",
+                  "$ref": "#/definitions/ContainerSettings"
+                },
+                "sparkConfigurations": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/SparkConfiguration"
+                  },
+                  "title": "Environment variables.",
+                  "description": "A collection of user defined Spark configurations to be set for spark cluster."
+                },
+                "environmentVariables": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/EnvironmentVariable"
+                  },
+                  "title": "Environment variables.",
+                  "description": "A collection of user defined environment variables to be set for spark cluster."
+                },
+                "driverSettings": {
+                  "title": "Spark driver settings.",
+                  "description": "Driver settings for Spark cluster.",
+                  "$ref": "#/definitions/SparkDriverSettings"
+                },
+                "workerSettings": {
+                  "title": "Spark worker settings.",
+                  "description": "Worker settings for Spark cluster.",
+                  "$ref": "#/definitions/SparkWorkerSettings"
+                },
+                "idleSecondsBeforeAutoTermination": {
+                  "type": "integer",
+                  "format": "int32",
+                  "description": "Deallocate nodes in the cluster after it is inactive for this time in seconds. If not set, nodes in this cluster will not be automatically deallocated"
+                },
+                "userAccountCredentials": {
+                  "title": "User account credentials.",
+                  "description": "Credentials for an administrator user account that will be created on each compute node.",
+                  "$ref": "#/definitions/UserAccountCredentials"
+                },
+                "subnet": {
+                    "title": "Subnet.",
+                    "description": "Virtual network subnet resource ID the compute nodes belong to.",
+                    "$ref": "#/definitions/ResourceId"

```

Comment:
I assume a linked access check will be defined for this?

Reply:
yes. we have linked access check for this in our manifest file

Reply:
common pattern is to use Id suffix for such property names, so subnetId is preferred.